### ISSUE TITLE
Crash on NavigationViewController.progressDidChange

### ISSUE THREAD

**Issue by Sealos** [mapbox/mapbox-navigation-ios#1236]
**Mapbox Navigation SDK version: 0.14.1**

Hello, we're getting some reports about crashes on fabric, can't reproduce locally, but this didn't was not ocurring on 0.13.
Stacktrace below:

```
Fatal Exception: CALayerInvalidGeometry
CALayer position contains NaN: [nan 0]

0  CoreFoundation                 0x182bc7164 __exceptionPreprocess
1  libobjc.A.dylib                0x181e10528 objc_exception_throw
2  CoreFoundation                 0x182bc70ac -[NSException initWithCoder:]
3  QuartzCore                     0x186c14640 CA::Layer::set_position(CA::Vec2<double> const&, bool)
4  QuartzCore                     0x186c065ec -[CALayer setPosition:]
5  QuartzCore                     0x186c06ba8 -[CALayer setFrame:]
6  UIKit                          0x18c17b5c8 -[UIView(Geometry) setFrame:]
7  MapboxNavigation               0x105aea314 specialized ProgressBar.(setProgress(CGFloat, animated : Bool) -> ()).(closure #1) (Style.swift:390)
8  MapboxNavigation               0x105ae44b4 ProgressBar.(setProgress(CGFloat, animated : Bool) -> ()).(closure #1) (Style.swift)
9  MapboxNavigation               0x105a71d14 thunk (NavigationMapView.swift)
10 UIKit                          0x18c1b6980 +[UIView(UIViewAnimationWithBlocks) _setupAnimationWithDuration:delay:view:options:factory:animations:start:animationStateGenerator:completion:]
11 UIKit                          0x18c1cd3b8 +[UIView(UIViewAnimationWithBlocks) animateWithDuration:delay:options:animations:completion:]
12 MapboxNavigation               0x105a6f5a8 specialized NavigationViewController.progressDidChange(notification : NSNotification) -> () (NavigationViewController.swift:5)
13 MapboxNavigation               0x105a6d530 @objc NavigationViewController.progressDidChange(notification : NSNotification) -> ()
14 CoreFoundation                 0x182b5913c __CFNOTIFICATIONCENTER_IS_CALLING_OUT_TO_AN_OBSERVER__
15 CoreFoundation                 0x182b586dc _CFXRegistrationPost
16 CoreFoundation                 0x182b58440 ___CFXNotificationPost_block_invoke
17 CoreFoundation                 0x182bd5e24 -[_CFXNotificationRegistrar find:object:observer:enumerator:]
18 CoreFoundation                 0x182a8ed60 _CFXNotificationPost
19 Foundation                     0x1834bb348 -[NSNotificationCenter postNotificationName:object:userInfo:]
20 MapboxCoreNavigation           0x106108944 specialized RouteController.locationManager(CLLocationManager, didUpdateLocations : [CLLocation]) -> () (RouteController.swift:639)
21 MapboxCoreNavigation           0x1060fd7c4 @objc RouteController.locationManager(CLLocationManager, didUpdateLocations : [CLLocation]) -> () (RouteController.swift)
22 CoreLocation                   0x1891c57bc (null)
23 CoreLocation                   0x1891c501c (null)
24 CoreLocation                   0x1891ad6b4 (null)
25 CoreFoundation                 0x182b6f590 __CFRUNLOOP_IS_CALLING_OUT_TO_A_BLOCK__
26 CoreFoundation                 0x182b6ee60 __CFRunLoopDoBlocks
27 CoreFoundation                 0x182b6cb4c __CFRunLoopRun
28 CoreFoundation                 0x182a8cc58 CFRunLoopRunSpecific
29 GraphicsServices               0x184938f84 GSEventRunModal
30 UIKit                          0x18c1e55c4 UIApplicationMain
```

Line in question:
https://github.com/mapbox/mapbox-navigation-ios/blob/d8bc35c7ca18ce0620b64d9eb434b7583196649a/MapboxNavigation/Style.swift#L433

Let me know if I can help more by submitting more information.

Thanks you!

**Comment by frederoni** [comment]
Possibly fixed by https://github.com/mapbox/mapbox-navigation-ios/pull/1245

Please let us know if you see similar crashes when using v0.15.0

**Comment by bsudekum** [comment]
This fix is now live in v0.16.0.
